routes/api/contactsMethots.js

```diff
@@ -0,0 +1,108 @@
+const fs = require("fs").promises;
+const path = require("path");
+const contactsPath = path.join(__dirname, "db", "contacts.json");
+const Joi = require("joi");
+
+const schema = Joi.object({
+  name: Joi.string().alphanum().min(3).max(30).required(),
+  email: Joi.string().email({
+    minDomainSegments: 2,
+    tlds: { allow: ["com", "net"] },
+  }),
+  phone: Joi.string().alphanum().min(7).max(12).required(),
+}).with("username", "birth_year");
+
+async function listContacts() {
+  try {
+    const data = await fs.readFile(contactsPath, "utf-8");
+    const contacts = JSON.parse(data);
+    return contacts;
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+async function getContactById(contactId) {
+  try {
+    const data = await fs.readFile(contactsPath, "utf-8");
+    const contacts = JSON.parse(data);
+    const contact = contacts.find((c) => c.id === contactId);
+    if (contact === undefined) {
+      return `id ${contactId} not found`;
+    }
+    return contact;
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+async function removeContact(contactId) {
+  try {
+    const data = await fs.readFile(contactsPath, "utf-8");
+    const contacts = JSON.parse(data);
+    const contact = contacts.find((c) => c.id === contactId);
+    const updatedContacts = contacts.filter(
+      (contact) => contact.id !== contactId
+    );
+    if (contact === undefined) {
+      return `id ${contactId} not found`;
+    }
+    const updatedContactsJSON = JSON.stringify(updatedContacts);
+
+    await fs.writeFile(contactsPath, updatedContactsJSON);
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+async function addContact(name, email, phone) {
+  const validation = schema.validate({ name, email, phone });
+  try {
+    if (validation.error) {
+      console.log(validation.error.message);
+      throw new Error(`Validation error: ${validation.error.message}`);
+    }
+    const data = await fs.readFile(contactsPath, "utf-8");
+    const contacts = JSON.parse(data);
+    const lastContact = contacts[contacts.length - 1];
+    const newId = parseInt(lastContact.id) + 1;
+    const newContact = { id: newId.toString(), name, email, phone };
+    contacts.push(newContact);
+    await fs.writeFile(contactsPath, JSON.stringify(contacts));
+    return newContact;
+  } catch (error) {
+    console.error(error);
+    return `Validation error: ${validation.error.message}`;
```

Comment:
Tu już pisałem że do zmiany, ale tu tak dla informacji powiem że obsługa tego try catcha nie zadziała jak powinien. Jeśli validacja będzie pomyślna a wystąpi błąd np w zapisie pliku to zwrócisz 201 z błędem jako string (i chyba nawet wyskoczy ci błąd przy res.status(201).json("Validation error.......")  bo to string a nie json)